libs/core/langchain_core/tools.py

```diff
@@ -89,19 +123,83 @@ def _create_subset_model(
             if field.required and not field.allow_none
             else Optional[field.outer_type_]
         )
+        if descriptions and field_name in descriptions:
+            field.field_info.description = descriptions[field_name]
         fields[field_name] = (t, field.field_info)
+
     rtn = create_model(name, **fields)  # type: ignore
+    rtn.__doc__ = textwrap.dedent(fn_description or model.__doc__ or "")
     return rtn
 
 
 def _get_filtered_args(
     inferred_model: Type[BaseModel],
     func: Callable,
+    *,
+    filter_args: Sequence[str],
 ) -> dict:
     """Get the arguments from a function's signature."""
     schema = inferred_model.schema()["properties"]
     valid_keys = signature(func).parameters
-    return {k: schema[k] for k in valid_keys if k not in ("run_manager", "callbacks")}
+    return {
+        k: schema[k]
+        for i, (k, param) in enumerate(valid_keys.items())
+        if k not in filter_args and (i > 0 or param.name not in ("self", "cls"))
+    }
+
+
+def _parse_python_function_docstring(function: Callable) -> Tuple[str, dict]:
```

Comment:
My opinion for the parser is that the default would've ideally been opt-in from the get-go AND it would've raised exceptions on incomplete doc-string.

Given we can't do that right now are we sure that we want opt-in by default? Instead of adding an option to fail loudly?